### PR TITLE
People2

### PR DESCRIPTION
**Pull request by shahnozayadgar**
- Write a one-line summary of what you did.
- Add a screenshot/video of the feature you developed (if applicable).

Created the initial people's page. This is meant to replace the original People pull request.

### CODE REVIEW

File: public/members/daeun.jpg
Comment (jhw123) [inline public/members/daeun.jpg:1]: DaEun updated her photo recently. Could you reflect that?

File: src/app/people/page.tsx [hunk 1]
```diff
@@ -1,5 +1,200 @@
-import { MEMBERS } from '@/data/members'
+'use client'
+
+import styled from '@emotion/styled'
+import { useState, useEffect } from 'react'
+//import { usePathname, useSearchParams } from 'next/navigation'
```
Comment (jhw123) [inline src/app/people/page.tsx:5]: please remove commented lines

File: src/app/people/page.tsx [hunk 2]
```diff
@@ -1,5 +1,200 @@
...
+const SpecialThanksSection = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 16px;
+`
+const SpecialThanksTitle = styled.h2`
+  font-size: 1.5rem;
+  color: ${Color.gray900};
+`
+const SpecialThanksImage = styled.img`
```
Comment (jhw123) [inline src/app/people/page.tsx:75]: we better use the `Image` component from next.js for optimization.

File: src/components/AlumniCard.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+          <Period>{mem.period}</Period>
+          {mem.currentPosition && <CurrentPosition>{mem.currentPosition}</CurrentPosition>}
+          {mem.education && <Education>{mem.education}</Education>}
+        </TextRow>
+      </Text>
+    </AlumniCardContainer>
+  )
+}
+
+export default AlumniCard
```
Comment (jhw123) [inline src/components/AlumniCard.tsx:84]: please use named export instead of default export if possible

File: src/data/members.ts [hunk 4]
```diff
@@ -66,16 +63,14 @@ export const MEMBER = {
     career: 'Ph.D. Student',
     img: 'hyunwoo.png',
     site: 'https://hyunwoo.me/',
-    thesis: 'https://kixlab.github.io/website-files/theses/thesis-ms-2019-hyunwoo.pdf',
   },
   seoyoungkim: {
-    firstName: 'Seoyoung',
+    firstName: 'Seyoung',
```
Comment (jhw123) [inline src/data/members.ts:68]: typo?

**Review by jhw123 — CHANGES_REQUESTED** [review]

**Comment by jhw123** [comment]
If we are not going to show Alumni's profile images, I think we better remove them in the repo

**Comment by jhw123** [comment]
Please update the screenshots!

File: src/app/people/page.tsx [hunk 5]
```diff
@@ -1,5 +1,200 @@
...
+          <SpecialThanksSection>
+            <SpecialThanksImage src="images/jura.jpeg" alt="Jura Coffee Machine" />
+            <SpecialThanksText>
+              <SpecialThanksSubtitle>Jura</SpecialThanksSubtitle>
+              <SpecialThanksDescription>Coffee Machine</SpecialThanksDescription>
+              <SpecialThanksDescription>
+                “Thank you for your steadfast warmth and the delightful brews that kickstart my mornings”
+              </SpecialThanksDescription>
+            </SpecialThanksText>
+          </SpecialThanksSection>
```
Comment (CheddarChoi) [inline src/app/people/page.tsx:175]:
```suggestion
<Divider />
          <Section key="thanks">
            <SectionTitle id="thanks">Special Thanks</SectionTitle>
            <SpecialThanksSection>
              <SpecialThanksImage src="images/jura.jpeg" alt="Jura Coffee Machine" />
              <SpecialThanksText>
                <SpecialThanksSubtitle>Jura</SpecialThanksSubtitle>
                <SpecialThanksDescription>Coffee Machine</SpecialThanksDescription>
                <SpecialThanksDescription>
                  <i>“Thank you for your steadfast warmth and the delightful brews that kickstart my mornings”</i>
                </SpecialThanksDescription>
              </SpecialThanksText>
            </SpecialThanksSection>
          </Section>
```

For consistency, it would be better to use Section component as much as possible.

File: src/app/people/page.tsx [hunk 6]
```diff
@@ -1,5 +1,200 @@
...
+  color: ${props => (props.selected ? Color.orange900 : Color.gray900)};
+
+  &:hover {
+    color: ${Color.orange900};
+  }
+`
+
+const SubCategoryTitle = styled.h2`
+  font-size: 1rem;
+  font-weight: bold;
```
Comment (CheddarChoi) [inline src/app/people/page.tsx:63]:
```suggestion
${FontVariant.title_sm}
```

For consistency, we need to use the pre-defined styles as much as possible. Please don't define any new font styles at least it is required for exceptional reasons.

File: src/app/people/page.tsx [hunk 7]
```diff
@@ -1,5 +1,200 @@
...
+`
+const SpecialThanksImage = styled.img`
+  width: 200px;
+  height: auto;
+  border-radius: 8px;
+`
+
+const SpecialThanksText = styled.div`
+  display: flex;
+  flex-direction: column;
```
Comment (CheddarChoi) [inline src/app/people/page.tsx:83]:
```suggestion
  flex-direction: column;
  gap: 4px;
```

Add gap here, instead of defining margin in SpecialThanksDescription

File: src/app/people/page.tsx [hunk 8]
```diff
@@ -1,5 +1,200 @@
...
+`
+
+const SpecialThanksSubtitle = styled.h3`
+  font-size: 1.25rem;
+  color: ${Color.orange900};
```
Comment (CheddarChoi) [inline src/app/people/page.tsx:88]:
```suggestion
  ${FontVariant.body_lg}
  color: ${Color.gray900};
```

Again, use a pre-defined font style for consistency. I changed color to match with the styles with the rest of the page.

File: src/app/people/page.tsx [hunk 9]
```diff
@@ -1,5 +1,200 @@
...
+  margin: 0;
+`
+
+const SpecialThanksDescription = styled.p`
+  font-size: 1rem;
+  color: ${Color.gray700};
+  margin: 0.5rem 0 0 0;
```
Comment (CheddarChoi) [inline src/app/people/page.tsx:95]:
```suggestion
  ${FontVariant.body_md}
  color: ${Color.gray700};
```

Again, use a pre-defined font style for consistency.

File: src/app/people/page.tsx
Comment (CheddarChoi) [inline src/app/people/page.tsx:1]: The ratio of images are not 1:1 in width of 1200 px ~ 1380 px.

**Review by CheddarChoi — CHANGES_REQUESTED** [review]

File: src/data/members.ts [hunk 10]
```diff
@@ -48,7 +47,6 @@ export const MEMBER = {
     career: 'Ph.D. Student',
     img: 'daeun.jpg',
     site: 'https://daeunchoi.com/',
-    thesis: 'https://kixlab.github.io/website-files/theses/thesis-ms-2024-daeun.pdf',
```
[the remark this answers is not in the document]
Comment (alextio) [reply src/data/members.ts:51]: As far as I remember, we agreed to not show the thesis link under each person's name.

File: src/app/people/page.tsx
Comment (shahnozayadgar) [reply src/app/people/page.tsx:1]: To what images are you referring here? Do you mean the images of members or ...?

Comment (CheddarChoi) [reply src/app/people/page.tsx:1]: Yes, the images of members.
스크린샷 2024-07-02 오후 3 06 56

File: src/data/members.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (CheddarChoi) [reply src/data/members.ts:51]: Then I think we need some other places to put it, for example, at the alumni section.

**Comment by CheddarChoi** [comment]
Also, the previous version of the website has a sorting algorithm for members. 

> {% if position.positionName contains 'Alumni' %}
    {% assign sorted_members = position.positionMembers | sort: 'leavingDate' | reverse %}
    {% elsif position.positionName contains 'Faculty and Postdocs' %}
    {% assign sorted_members = position.positionMembers %}
    {% else %}
    {% assign sorted_members = position.positionMembers | sort: 'lastName' | group_by: 'lastName'%} 
    {% endif %}

File: src/app/people/page.tsx [hunk 11]
```diff
@@ -1,5 +1,229 @@
...
+  useEffect(() => {
+    handleSidebarList(MEMBERS_KEY)
+  }, [MEMBERS_KEY])
+
+  const handleSidebarList = (list: string[]) => {
+    const sections = CareerTypes.filter(career => career !== 'Alumni').map(career =>
+      career.replace(/\s+/g, '').replace('.', '')
+    )
+    sections.push('alumni')
+    setSidebarList(sections)
```
Comment (alextio) [inline src/app/people/page.tsx:146]: Is there a reason why you formatted the text to have no whitespace in the Sidebar?

File: src/app/people/page.tsx [hunk 12]
```diff
@@ -1,5 +1,226 @@
...
+                    id={career.replace(/\s+/g, '').replace('.', '')}
+                    ref={el => {
+                      sectionRefs.current[career] = el
+                    }}
+                  >
+                    <SectionTitle id={career.replace(/\s+/g, '').replace('.', '')}>{career}</SectionTitle>
+                    <SectionContent>
+                      {MEMBERS_KEY.filter(key => MEMBERS[key].career == career && career != 'Alumni').map(key => (
+                        <MemberCard key={key} mem={MEMBERS[key]} />
+                      ))}
```
Comment (CheddarChoi) [inline src/app/people/page.tsx:175]: Related to my comment I left last time, we need to add a sorting algorithm that we have in the prev version, not just get the order from the MEMBERS variable, as it will be changed as people are added

**Comment by CheddarChoi** [comment]
I think we need to change Associate Professor --> Faculty

File: src/app/people/page.tsx [hunk 13]
```diff
@@ -1,5 +1,226 @@
...
+      sections.forEach(section => {
+        if (section) {
+          observer.unobserve(section)
+        }
+      })
+    }
+  }, [])
+
+  useEffect(() => {
+    handleSidebarList(MEMBERS_KEY)
```
Comment (jhw123) [inline src/app/people/page.tsx:135]: if you call this function once at the beginning, can't we just set the initial value of the sidebarList with the logic in the handleSidebarList?

File: src/app/people/page.tsx [hunk 14]
```diff
@@ -1,5 +1,226 @@
...
+                        <MemberCard key={key} mem={MEMBERS[key]} />
+                      ))}
+                    </SectionContent>
+                  </Section>
+                  <Divider />
+                </React.Fragment>
+              )
+          )}
+          <Section
+            key="alumni"
```
Comment (jhw123) [inline src/app/people/page.tsx:183]: why did you add key to this component?

File: src/components/AlumniCard.tsx [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+  color: ${Color.gray700};
+`
+const Education = styled.div`
+  font-size: 0.9rem;
+  display: flex;
+  gap: 6px;
+  justify-content: center;
+  align-items: center;
+  color: ${Color.gray700};
+  }
```
Comment (jhw123) [inline src/components/AlumniCard.tsx:58]:
```suggestion
```

File: src/components/AlumniCard.tsx [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+`
+
+export default function AlumniCard({ mem }: { mem: IMember }) {
+  return (
+    <AlumniCardContainer>
+      <Text>
+        <TextRow>
+          {mem.site ? (
+            <NameWithWebsite href={mem.site}>
+              {mem.firstName} {mem.lastName}
```
Comment (jhw123) [inline src/components/AlumniCard.tsx:68]: I think we have a full name function in the member class

File: src/components/ImageWithFallback.tsx [hunk 17]
```diff
@@ -0,0 +1,12 @@
+import { useState } from 'react'
+import Image, { ImageProps } from 'next/image'
+
+export const ImageWithFallback = ({ fallbackSrc, ...props }: { fallbackSrc: string } & ImageProps) => {
```
Comment (jhw123) [inline src/components/ImageWithFallback.tsx:4]: I am not sure how useful this component is for static websites

File: src/components/MemberCard.tsx [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+
+const Affiliation = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 6px;
+  color: ${Color.gray700};
+  }
+`
+
+export default function MemberCard({ mem }: { mem: IMember }) {
```
Comment (jhw123) [inline src/components/MemberCard.tsx:66]:
```suggestion
interface Props {
  mem: IMember
}

export default function MemberCard({ mem }: Props) {
```

Comment (jhw123) [inline src/components/MemberCard.tsx:66]: Let's be consistent with the exporting style (export default vs export const).
I prefer the latter.

**Review by jhw123 — DISMISSED** [review]

File: src/app/people/page.tsx (hunk 13 above)
Comment (alextio) [reply src/app/people/page.tsx]: Yes you're right. In fact, we don't even need a handleSidebarList function since the sections are fixed

File: src/app/people/page.tsx (hunk 12 above)
Comment (alextio) [reply src/app/people/page.tsx]: [2ae2871](https://github.com/kixlab/website/pull/40/commits/2ae28718c6c5186ec6e5be196ffe2576ed8fa3f4)
Had to change the sorting algorithm slightly for alumni, since not all of them have an end date. The updated algorithm sorts them by their start date.

File: src/components/ImageWithFallback.tsx (hunk 17 above)
Comment (alextio) [reply src/components/ImageWithFallback.tsx:4]: I'm not sure either, but it seems like a good safety net against any errors.

File: src/app/people/page.tsx (hunk 14 above)
Comment (alextio) [reply src/app/people/page.tsx:147]: Good question.

File: src/components/AlumniCard.tsx (hunk 16 above)
Comment (alextio) [reply src/components/AlumniCard.tsx:68]: We'll be using plain JSON objects, so no functions available.

File: src/components/Filter.tsx [hunk 19]
```diff
@@ -75,7 +75,7 @@ const FilterOption = styled.li`
   }
 `
 
-interface Props {
```
Comment (jhw123) [inline src/components/Filter.tsx:78]: It is unnecessary to change local type names. It makes the dev experience worse in many cases because it complicates copy and pasting between components

File: src/data/publications.ts [hunk 20]
```diff
@@ -121,7 +122,7 @@ export const PUBLICATIONS: Publication[] = [
   },
   {
     title: 'VIVID: Human-AI Collaborative Authoring of Vicarious Dialogues from Lecture Videos',
-    authors: ['Seulgi Choi', 'Hyewon Lee', 'Yoonjoo Lee', MEMBERS.juhokim],
```
[the remark this answers is not in the document]
Comment (alextio) [reply src/data/publications.ts:124]: Problem is: non-Kixlab authors are plain string types, and it's clunky to mix and match Member types with strings.

File: src/components/SideBar.tsx [hunk 21]
```diff
@@ -31,31 +31,73 @@ const SidebarLink = styled.a`
...
+}
+
+interface ISidebar {
+  handleLinkClick?: () => void
   sidebarList: string[]
+  sectionRefs: React.RefObject<Record<string, HTMLElement | null>>
+  observerOptions?: IobserverOptions
 }
 
 const capitalizeWords = (s: string) => {
```
[the remark this answers is not in the document]
Comment (alextio) [reply src/components/SideBar.tsx:55]: [4e49ce7](https://github.com/kixlab/website/pull/40/commits/4e49ce794d28db34102d9644cdf78275e3c033da)
One small issue I noticed is that we now have 'M S Student' and 'Ph D student' instead of 'M S. Student' and 'Ph D. Student'.

**Review by alextio — APPROVED** [review]
